src/components/Catalog.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react'
+import Movie from './Movie'
+import '../styles/catalog.css';
+
+class Catalog extends Component {
+
+    constructor() {
+        super()
+
+        this.state = {
+            rentedMovies: [],
+            budget: 10 
+        }
+    }
+
+    rentIt = id => {
+        if(this.state.budget > 3){
+            let arr = [...this.state.rentedMovies]
+            arr.push(id)
+            this.setState({ rentedMovies: arr }, function () {
+                this.setState({ budget: (this.state.budget - 3) })
+            })
+        }
+
+    }
+    refound = id => {
+        // this.props.method(id)
+        let arr = [...this.state.rentedMovies]
+        for (let i in arr) {
+            if (arr[i].id === id) arr.splice(i, 0)
+        }
+        this.setState({ rentedMovies: arr }, function () {
+            this.setState({ budget: (this.state.budget + 3) })
+        })
+    }
+
+    rent = id => this.props.method(id)
+
+    handleSearch = e => this.props.search(e.target.value)
+
+    renderRented = (moviesArr) => {
+        return (
+            <div>
+                <div>rents:</div>
+                {moviesArr.filter(m => m.title.toLowerCase().search(this.props.searchVal) !== -1).map((m, i) => { return (<Movie key={i} movie={m} method={this.rent} rentIt={this.rentIt} refound={this.refound} budget={this.state.budget} />) })}
+            </div>
+        )
+    }
+
+    render() {
+        let movies = this.props.movies
```

Comment:
`const {movies, rentedMovies} = this.props`

deconstruct it like this. All is defined in one line